test/xexp.jl

```diff
@@ -0,0 +1,39 @@
+@testset "xexp" begin
+    @test isnan(xexp(NaN))
+    @test isinf(xexp(Inf))
+    @test xexp(-Inf) == 0
+    @test xexp(0.0) == 1
+
+    @test isinf(xexp(710.0))
+    @test xexp(-746.0) == 0
+
+    x = linspace(708.4, 709.7, 25)
+    @test_approx_eq xexp.(x) exp.(x)
```

Comment:
The source code appears to [claim](https://github.com/JuliaLang/julia/blob/c48d203702f7b439792ffd9d24abd23e47de171b/base/test.jl#L835) `@test_approx_eq` to be "legacy," which is why I wanted Stefan's input.